src/targets/__tests__/symbolCollector.test.ts

```diff
@@ -0,0 +1,146 @@
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+  });
+
+  test('config missing', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >) = jest.fn();
+
+    expect(getSymbolCollectorInstance).toThrowErrorMatchingInlineSnapshot(
+      `"Required configuration not found in configuration file. See the documentation for more details."`
+    );
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+  });
+
+  test('symbol collector present and config ok', () => {
+    (checkExecutableIsPresent as jest.MockedFunction<
+      typeof checkExecutableIsPresent
+    >) = jest.fn();
+
+    const symCollector = getSymbolCollectorInstance(customConfig);
+    const actualConfig = symCollector.symbolCollectorConfig;
+    expect(checkExecutableIsPresent).toHaveBeenCalledTimes(1);
+    expect(checkExecutableIsPresent).toHaveBeenLastCalledWith(
+      SYM_COLLECTOR_BIN_NAME
+    );
+    expect(actualConfig).toHaveProperty('serverEndpoint');
+    expect(actualConfig).toHaveProperty('batchType');
+    expect(actualConfig).toHaveProperty('bundleIdPrefix');
+  });
+});
+
+describe('publish', () => {
+  test('no artifacts found', () => {
+    const symCollector = getSymbolCollectorInstance(customConfig);
+    symCollector.getArtifactsForRevision = jest
+      .fn()
+      .mockReturnValueOnce(() => []);
+    expect(spawnProcess).not.toHaveBeenCalled();
+  });
+
+  test('with artifacts', async () => {
+    (withTempDir as jest.MockedFunction<typeof withTempDir>).mockImplementation(
+      async cb => await cb('tmpDir')
+    );
+    (spawnProcess as jest.MockedFunction<
+      typeof spawnProcess
+    >).mockImplementation(() => Promise.resolve(undefined));
+
+    const mockedArtifacts = [
+      { filename: 'artifact1', storedFile: { lastUpdated: 'tomorrow' } },
+      { filename: 'artifact2', storedFile: { lastUpdated: 'next week' } },
+      { filename: 'artifact2', storedFile: { lastUpdated: 'in 5 years' } },
+    ];
+
+    const symCollector = getSymbolCollectorInstance(customConfig);
+    symCollector.getArtifactsForRevision = jest
+      .fn()
+      .mockReturnValueOnce(mockedArtifacts);
+    symCollector.artifactProvider.downloadArtifact = jest.fn();
+
+    await symCollector.publish('version', 'revision');
+
+    expect(symCollector.getArtifactsForRevision).toHaveBeenCalledTimes(1);
+    expect(
+      symCollector.artifactProvider.downloadArtifact
+    ).toHaveBeenCalledTimes(mockedArtifacts.length);
+
+    expect(spawnProcess).toHaveBeenCalledTimes(1);
+    const [cmd, args] = (spawnProcess as jest.MockedFunction<
+      typeof spawnProcess
+    >).mock.calls[0] as string[];
+    expect(cmd).toBe(SYM_COLLECTOR_BIN_NAME);
+    expect(args).toMatchInlineSnapshot(`
```

Comment:
Why inline snapshot over just asserting on the array?

Reply:
Snapshots are much easier to manage 🙂